todo/src/filestorage.rs

```diff
@@ -0,0 +1,64 @@
+pub(crate) mod error;
+use crate::todo::Todo;
+use crate::todos::Todos;
+use async_trait::async_trait;
+use error::StorageError;
+use tokio::fs::{read_to_string, write};
+
+pub struct FileStorage {
+    pub path: String,
+}
+
+#[async_trait]
+pub trait Storage {
+    async fn get_todos_from_filestorage(&mut self) -> Result<Todos, StorageError>;
+    async fn write_filestorage(&mut self, todo_list: &mut Todos) -> Result<(), StorageError>;
+}
+
+#[async_trait]
+impl Storage for FileStorage {
+    async fn get_todos_from_filestorage(&mut self) -> Result<Todos, StorageError> {
+        let mut todo_vec = Vec::new();
+        let todo_str = read_to_string(&self.path)
+            .await
+            .map_err(|_| StorageError::Read)?;
+
+        let mut vec_todo: Vec<&str> = todo_str.split('\n').collect();
+        vec_todo.truncate(vec_todo.len() - 1);
+
+        for line in vec_todo.iter() {
+            let todo = FileStorage::build_todo(line.to_string())?;
+            todo_vec.push(todo);
+        }
```

Comment:
Para evitar duas alocações aqui, você pode tentar fazer tudo em termos da API de Iteradores. Provavelmente terá que usar um `map` .

Talvez você também precise de usar a implementação de `collect` que, dado um iterador de `Result<T, E>`, te dê um `Result<Vec<T>, E>` (ao invés de `Vec<Result<T, E>>`). Assim, você só usa a `?` uma vez, e o `collect` retorna `Err` se o `map` encontrar qualquer `Err` ao invés de `Ok`.

[Ref.](https://doc.rust-lang.org/std/iter/trait.FromIterator.html#impl-FromIterator%3CResult%3CA%2C%20E%3E%3E-for-Result%3CV%2C%20E%3E)